
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>pandas.core.window.Rolling.corr &#8212; Intel® Scalable Dataframe Compiler 0.1 documentation</title>
    <link rel="stylesheet" href="../_static/sdc.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="../_static/sidebar.js"></script>
    <script type="text/javascript" src="../_static/copybutton.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link href='https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,600' rel='stylesheet' type='text/css'/>

  </head><body>
<div class="topbar">
  <a class="brand" title="Intel Python projects" href="../other_intel_python_projects.html"></a>
  <a class="brand_sdc" title="Documentation Home" href="../index.html"></a>

  <ul>
    <li><a class="exampleslink" title="Examples" href="../examples.html"></a></li>
    <li><a class="issueslink" title="Issues" href="https://software.intel.com/en-us/distribution-for-python/support/get-help"></a></li>
    <li><a class="emaillink" title="Email" href="mailto:scripting@intel.com"></a></li>
    <li><a class="homelink" title="SDC GitHub" href="https://github.com/IntelPython/sdc"></a></li>
    <li>
      
      
<form action="../search.html" method="get">
  <input type="text" name="q" placeholder="Search" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
      
    </li>
  </ul>
</div>

<div class="related">
    <h3>Navigation</h3>
    <ul>
      <li>
	<a href="../index.html">Intel® Scalable Dataframe Compiler 0.1 documentation</a>
	 &#187;
      </li>
      
       
    </ul>
</div>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="id1">
<h1>pandas.core.window.Rolling.corr<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h1>
<p>Calculate rolling correlation.</p>
<dl class="field-list simple">
<dt class="field-odd">param other</dt>
<dd class="field-odd"><dl class="simple">
<dt>Series, DataFrame, or ndarray, optional</dt><dd><p>If not supplied then will default to self.</p>
</dd>
</dl>
</dd>
<dt class="field-even">param pairwise</dt>
<dd class="field-even"><dl class="simple">
<dt>bool, default None</dt><dd><p>Calculate pairwise combinations of columns within a
DataFrame. If <cite>other</cite> is not specified, defaults to <cite>True</cite>,
otherwise defaults to <cite>False</cite>.
Not relevant for <a class="reference external" href="https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v1.0.1)"><code class="xref py py-class docutils literal notranslate"><span class="pre">Series</span></code></a>.
**kwargs
Unused.</p>
</dd>
</dl>
</dd>
<dt class="field-odd">return</dt>
<dd class="field-odd"><p>Series or DataFrame
Returned object type is determined by the caller of the
rolling calculation.</p>
</dd>
</dl>
<div class="section" id="limitations">
<h2>Limitations<a class="headerlink" href="#limitations" title="Permalink to this headline">¶</a></h2>
<p>Series elements cannot be max/min float/integer. Otherwise SDC and Pandas results are different.
Resulting Series has default index and name.</p>
</div>
<div class="section" id="examples">
<h2>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h2>
<div class="literal-block-wrapper docutils container" id="ex-series-rolling-corr">
<div class="code-block-caption"><span class="caption-text">Calculate rolling correlation.</span><a class="headerlink" href="#ex-series-rolling-corr" title="Permalink to this code">¶</a></div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">numba</span> <span class="kn">import</span> <span class="n">njit</span>


<span class="nd">@njit</span>
<span class="k">def</span> <span class="nf">series_rolling_corr</span><span class="p">():</span>
    <span class="n">series</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">([</span><span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">8</span><span class="p">])</span>  <span class="c1"># Series of 3, 3, 3, 5, 8</span>
    <span class="n">other</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">([</span><span class="mi">3</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">8</span><span class="p">])</span>  <span class="c1"># Series of 3, 4, 4, 4, 8</span>
    <span class="n">out_series</span> <span class="o">=</span> <span class="n">series</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="mi">4</span><span class="p">)</span><span class="o">.</span><span class="n">corr</span><span class="p">(</span><span class="n">other</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">out_series</span>  <span class="c1"># Expect series of NaN, NaN, NaN, 0.333333, 0.916949</span>


<span class="nb">print</span><span class="p">(</span><span class="n">series_rolling_corr</span><span class="p">())</span>
</pre></div>
</div>
</div>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ python ./series/rolling/series_rolling_corr.py
Traceback (most recent call last):
  File &quot;./series/rolling/series_rolling_corr.py&quot;, line 40, in &lt;module&gt;
    print(series_rolling_corr())
  File &quot;/localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/dispatcher.py&quot;, line 401, in _compile_for_args
    error_rewrite(e, &#39;typing&#39;)
  File &quot;/localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/dispatcher.py&quot;, line 344, in error_rewrite
    reraise(type(e), e, None)
  File &quot;/localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py&quot;, line 668, in reraise
    raise value.with_traceback(tb)
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function hpat_pandas_series_rolling_corr at 0x7fc76e31a320&gt;) with argument(s) of type(s): (SeriesRollingType(series(int64, array(int64, 1d, C), none, False), win_type=none, on=none, closed=none), series(int64, array(int64, 1d, C), none, False))
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function arr_corr at 0x7fc76e57a680&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C))
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function corrcoef at 0x7fc78540b290&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C))
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function corrcoef at 0x7fc78540b290&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py (108)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py&quot;, line 108:
def arr_corr(x, y):
    &lt;source elided&gt;

    return numpy.corrcoef(x, y)[0, 1]
    ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function corrcoef at 0x7fc78540b290&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C))
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function corrcoef at 0x7fc78540b290&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py (108)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py&quot;, line 108:
def arr_corr(x, y):
    &lt;source elided&gt;

    return numpy.corrcoef(x, y)[0, 1]
    ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function arr_corr at 0x7fc76e57a680&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py (482)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py&quot;, line 482:
        def calc_corr(main, other, minp):
            &lt;source elided&gt;
            else:
                return arr_corr(main[valid], other[valid])
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function arr_corr at 0x7fc76e57a680&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C))
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function corrcoef at 0x7fc78540b290&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C))
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function corrcoef at 0x7fc78540b290&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py (108)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py&quot;, line 108:
def arr_corr(x, y):
    &lt;source elided&gt;

    return numpy.corrcoef(x, y)[0, 1]
    ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function corrcoef at 0x7fc78540b290&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C))
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function cov at 0x7fc78540b050&gt;) with argument(s) of type(s): (array(int64, 1d, C), array(int64, 1d, C), bool)
 * parameterized
In definition 0:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
In definition 1:
    TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;) with argument(s) of type(s): (array(float64, 2d, C), bool, none)
 * parameterized
In definition 0:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
In definition 1:
    LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
scipy 0.16+ is required for linear algebra

File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2419:
def np_cov_impl_inner(X, bias, ddof):
    &lt;source elided&gt;
    # calculate result - requires blas
    c = np.dot(X, np.conj(X.T))
    ^

[1] During: lowering &quot;$76call_method.23 = call $62load_method.16(X.1, $74call_method.22, func=$62load_method.16, args=[Var(X.1, arraymath.py:2416), Var($74call_method.22, arraymath.py:2419)], kws=(), vararg=None)&quot; at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2419)
    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/six.py:669
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function np_cov_impl_inner at 0x7fc77110c950&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2610)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2610:
        def np_cov_impl(m, y=None, rowvar=True, bias=False, ddof=None):
            &lt;source elided&gt;
            else:
                return np_cov_impl_inner(X, bias, ddof)
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function cov at 0x7fc78540b050&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py (2642)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/targets/arraymath.py&quot;, line 2642:
        def np_corrcoef_impl(x, y=None, rowvar=True):
            c = np.cov(x, y, rowvar)
            ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function corrcoef at 0x7fc78540b290&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py (108)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py&quot;, line 108:
def arr_corr(x, y):
    &lt;source elided&gt;

    return numpy.corrcoef(x, y)[0, 1]
    ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(&lt;function arr_corr at 0x7fc76e57a680&gt;)
[2] During: typing of call at /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py (482)


File &quot;../../miniconda3/envs/sdc_env/lib/python3.7/site-packages/sdc/datatypes/hpat_pandas_series_rolling_functions.py&quot;, line 482:
        def calc_corr(main, other, minp):
            &lt;source elided&gt;
            else:
                return arr_corr(main[valid], other[valid])
                ^

    raised from /localdisk/vssmirno/miniconda3/envs/sdc_env/lib/python3.7/site-packages/numba/typeinfer.py:985
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: BoundFunction((&lt;class &#39;sdc.datatypes.hpat_pandas_series_rolling_types.SeriesRollingType&#39;&gt;, &#39;corr&#39;) for SeriesRollingType(series(int64, array(int64, 1d, C), none, False), win_type=none, on=none, closed=none))
[2] During: typing of call at ./series/rolling/series_rolling_corr.py (35)


File &quot;series/rolling/series_rolling_corr.py&quot;, line 35:
def series_rolling_corr():
    &lt;source elided&gt;
    other = pd.Series([3, 4, 4, 4, 8])  # Series of 3, 4, 4, 4, 8
    out_series = series.rolling(4).corr(other)
    ^
</pre></div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<dl class="simple">
<dt><a class="reference internal" href="pandas.Series.rolling.html#pandas-series-rolling"><span class="std std-ref">Series.rolling</span></a></dt><dd><p>Calling object with a Series.</p>
</dd>
<dt><a class="reference internal" href="pandas.DataFrame.rolling.html#pandas-dataframe-rolling"><span class="std std-ref">DataFrame.rolling</span></a></dt><dd><p>Calling object with a DataFrame.</p>
</dd>
<dt><a class="reference internal" href="pandas.Series.corr.html#pandas-series-corr"><span class="std std-ref">Series.corr</span></a></dt><dd><p>Similar method for Series.</p>
</dd>
<dt><a class="reference internal" href="pandas.DataFrame.corr.html#pandas-dataframe-corr"><span class="std std-ref">DataFrame.corr</span></a></dt><dd><p>Similar method for DataFrame.</p>
</dd>
</dl>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper"><h3>Table of Contents</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">What is Intel® SDC?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started.html">Getting Started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../compilation.html">Compiling With Intel® SDC</a></li>
<li class="toctree-l1"><a class="reference internal" href="../performance.html">Getting Performance With Intel® SDC</a></li>
<li class="toctree-l1"><a class="reference internal" href="../performance.html#compilation-overheads">Compilation Overheads</a></li>
<li class="toctree-l1"><a class="reference internal" href="../performance.html#automatic-parallelization">Automatic Parallelization</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflow.html">Step by step on typical data analysis workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../apireference.html">API Reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../examples.html">List of examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="../for_developers.html">Contribute to Intel® Scalable Dataframe Compiler Development</a></li>
<li class="toctree-l1"><a class="reference internal" href="../other_intel_python_projects.html">Other Intel projects for high-performance Python</a></li>
</ul>


<form action="../search.html" method="get">
  <input type="text" name="q" placeholder="Search" />
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
<footer class="footer">
  <p class="pull-right"> &nbsp;
    <a href="#">Back to Top</a></p>
  <p>
    &copy; Copyright 2019, Intel Corporation.<br/>
    Created using <a href="http://www.sphinx-doc.org/en/stable/">Sphinx</a> 2.4.0. &nbsp;
  </p>
</footer>
  </body>
</html>